src/main/java/addon/Tasklist.java

```diff
@@ -0,0 +1,226 @@
+package addon;
+
+import java.time.LocalDateTime;
+
+import java.util.ArrayList;
+
+import addon.Ui.IncorrectFormatException;
+
+/**
+ * Handles methods involving the Arraylists of current tasks, as well as containing the Task classes
+ */
+public class Tasklist {
+
+    private final ArrayList<Task> list;
+
+    public Tasklist(ArrayList<Task> list) {
+        this.list = list;
+    }
+
+    /**
+     * Adds Task entry to the Arraylist
+     *
+     * @param args Array of strings representing in order: type of task, name of task.
+     * @param date LocalDateTime object involved with the task, left as an arbitrary null date if Todotask.
+     */
+    public void addEntry(String[] args, LocalDateTime date) {
+        if (args[0].equals("T")) {
+            addon.Tasklist.Todo add = new addon.Tasklist.Todo(args[1]);
+            this.list.add(add);
+            System.out.println(Ui.BAR + "\n    New todo added: " + args[1] + "\n\n    Now you have " + this.list.size()
+                    + " tasks in your list\n" + Ui.BAR);
+            Storage.appendFile(add);
+        } else if (args[0].equals("E")) {
+            Event add = new Event(args[1], date);
+            this.list.add(add);
+            System.out.println(Ui.BAR + "\n    New event added: " + args[1] + " on " + Ui.printDate(date, true)
+                    + "\n\n    Now you have " + this.list.size() + " tasks in your list\n" + Ui.BAR);
+            Storage.appendFile(add);
+        } else {
+            Deadline add = new Deadline(args[1], date);
+            this.list.add(add);
+            System.out.println(Ui.BAR + "\n    New deadline added: " + args[1] + " by " + Ui.printDate(date, true)
+                    + "\n\n    Now you have " + this.list.size() + " tasks in your list\n" + Ui.BAR);
+            Storage.appendFile(add);
+        }
+    }
+    /**
+     * Removes specified task entry from the Arraylist
+     *
+     * @param num Index number of the item to be removed.
+     */
+    public void removeEntry(int num) throws IncorrectFormatException {
+        if (num >= 1 && num <= this.list.size() + 1) {
+            System.out.println(Ui.BAR + "\n    Nice! I've removed this task off the face of the Earth:\n\n    "
+                    + (list.get(num - 1)).toString() + "\n    Now you have " + list.size() +
+                    " tasks in the list.\n" + Ui.BAR);
+            this.list.remove(num - 1);
+            Storage.rewriteFile(this.list);
+        } else {
+            throw new IncorrectFormatException("List does not contain this item number. Try again?");
+        }
+    }
+
+    /**
+     * Changes the "Done" status of the specified task entry from the Arraylist
+     *
+     * @param num Index number of the item to be changed.
+     */
+    public void changeDone(int num) throws IncorrectFormatException {
+        if (num >= 1 && num <= this.list.size() + 1) {
```

Comment:
try-catching a IndexOutOfBoundsException could work too. That way you can possibly print out the error as well for more information.